src/wp-includes/class-wp-query.php

```diff
@@ -4755,6 +4722,55 @@ public function generate_postdata( $post ) {
 
 		return $elements;
 	}
+
+	/**
+	 * Generate cache key.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param array $cache_args
+	 * @param string $fields
+	 *
+	 * @return string
+	 */
+	protected function generate_cache_key( array $cache_args, $request ) {
+		global $wpdb;
+
+		unset(
+			$cache_args['suppress_filters'],
+			$cache_args['cache_results'],
+			$cache_args['fields'],
+			$cache_args['update_post_meta_cache'],
+			$cache_args['update_post_term_cache'],
+			$cache_args['lazy_load_term_meta'],
+			$cache_args['update_menu_item_cache']
+		);
+
+		$placeholder = $wpdb->placeholder_escape();
+		foreach ( $cache_args as $key => $value ) {
+			if ( is_array( $value ) ) {
+				foreach ( $value as $n_key => $n_value ) {
+					if ( is_string( $n_value ) && str_contains( $n_value, $placeholder ) ) {
+						$cache_args[ $key ][ $n_key ] = $wpdb->remove_placeholder_escape( $n_value );
+					}
+				}
+			}
```

Comment:
This should probably use a recursive function.

The current case supports `$cache_args[][]` but not ``$cache_args[][][]` or deeper which is possible with meta and taxo queries

Reply:
Or perhaps try using `array_walk_recursive()`, would probably be a little faster too.

Reply:
Fixed.